src/query/functions/utils/unique.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package utils
+
+// UniqueCount calculates the number of unique strings in a given slice.
+func UniqueCount(names []string) int {
+	uniqueMap := make(map[string]struct{}, len(names))
+	for _, s := range names {
+		uniqueMap[s] = struct{}{}
+	}
+
+	return len(uniqueMap)
```

Comment:
Per Nikunj's previous suggestion, maybe just return `uniqueMap` and let the caller get `len`?

Reply:
Could be valid; maybe have 2 methods, one which just returns len, other returns the map (or better yet, the map flattened back to a []string)